app/src/main/java/com/ciandt/dragonfly/example/features/feedback/FeedbackPresenter.kt

```diff
@@ -37,40 +40,76 @@ class FeedbackPresenter(
         }
     }
 
-    override fun setClassifications(classifications: List<Classifier.Classification>) {
 
-        results.clearAndAddAll(classifications)
+    override fun setClassifications(classifications: LinkedHashMap<String, ArrayList<Classifier.Classification>>) {
+        results.clear()
+        results.putAll(classifications)
 
-        if (results.isEmpty()) {
+        val labels: LinkedList<Pair<String, Int>> = LinkedList()
+
+        classifications.entries.forEach { entry ->
+
+            val list = entry.value
+            if (list.isEmpty()) {
+                labels.add(Pair.create("", 0))
+                return
+            }
+
+            if (!list[0].hasTitle()) {
+                labels.add(Pair.create("", 0))
+                return
+            }
+
+            labels.add(Pair.create(list[0].title, formatConfidence(list[0].confidence)))
+        }
+
+        view?.showMainClassifications(labels)
+        
+
+        oldResults.clearAndAddAll(classifications.entries.first().value)
+
+        if (oldResults.isEmpty()) {
             view?.showNoClassifications()
             return
         }
 
         if (userFeedback == null) {
-            view?.showClassifications(results.head().title, results.tail())
+
+            val displayResults = LinkedHashMap<String, ArrayList<Classifier.Classification>>()
+            results.forEach { (key, value) ->
+                val index = results.keys.indexOf(key)
+                val title = if (index >= 0) {
+                    model.outputDisplayName[index]
+                } else {
+                    key
+                }
+                displayResults.put(title, value)
+            }
+
+            view?.showClassifications(displayResults)
         } else {
             userFeedback!!.let {
                 if (it.isPositive()) {
-                    view?.showPositiveClassification(it.actualLabel, results.tail(), false)
+                    view?.showPositiveClassification(it.actualLabel, oldResults.tail(), false)
```

Comment:
Why oldResults (not sure it's the same code as the one I asked this question in the other PR) here and below?